modules/cluster/main.tf

```diff
@@ -66,6 +66,9 @@ module "eks" {
       asg_desired_capacity = var.desired_node_count
       asg_min_size         = var.min_node_count
       asg_max_size         = var.max_node_count
+      if var.enable_spot_instances {
```

Comment:
I dont think this will work, terraform12 does not support if clause in this context.

Reply:
`spot_price           = (var.enable_spot_instances ? var.spot_price : null)`